### PR TITLE
Local Pairing update for `StopNode`

### LINKED ISSUE #3399: Release Messenger After Local Pairing Has Completed
During the Local Pairing process `backend.StartNodeWithAccount` is called:

```go
	activeAccount, _ := s.backend.GetActiveAccount()
	if activeAccount == nil { // not login yet
		s.backend.UpdateRootDataDir(nodeConfig.RootDataDir)
		// because client don't know keyUID before received data, we need help client to update keystore dir
		keystoreDir := filepath.Join(nodeConfig.KeyStoreDir, account.KeyUID)
		nodeConfig.KeyStoreDir = keystoreDir
		if accountPayload.exist {
			err = s.backend.StartNodeWithAccount(*account, accountPayload.password, nodeConfig)
		} else {
			accountManager := s.backend.AccountManager()
			err = accountManager.InitKeystore(filepath.Join(nodeConfig.RootDataDir, keystoreDir))
			if err != nil {
				return err
			}
			rmp.setting.InstallationID = nodeConfig.ShhextConfig.InstallationID
			rmp.setting.CurrentNetwork = settingCurrentNetwork

			err = s.backend.StartNodeWithAccountAndInitialConfig(*account, accountPayload.password, *rmp.setting, nodeConfig, rmp.subAccounts)
		}
		if err != nil {
			return err
		}
	}
```

This process effectively logs in the user. However, despite the Local Pairing process requiring a fully qualified instance of `Messenger`, the responsibility of Local Pairing should not be to log the application in **for the user**. Instead to maintain segregation of responsibility the Local Pairing process flow should clean up the logged in state of the application and log out once all Local Pairing functionality has completed. After the Local Pairing process flow the endpoint / application / user should choose to login.

The wider application should not have an expectation and/or rely on Local Pairing to guarantee a logged in state.

Consider the implications of :

- https://github.com/status-im/status-go/issues/3241
- https://github.com/status-im/status-go/issues/3304
- https://github.com/status-im/status-go/issues/3306

cc @igor-sirotin

### PR DESCRIPTION
**Pull request by igor-sirotin**
Closes https://github.com/status-im/status-go/issues/3399

This is an continuation of https://github.com/status-im/status-go/pull/3525
With those updated manual sign in works fine in Desktop and Mobile 👌 

### 1. Replace `StopNode` with `LogOut`

Calling `Login` after `StopNode` was failing because `appDB` wasn't create correctly, as it wasn't cleared. So whole DB initialization steps were just skipped here:
https://github.com/status-im/status-go/blob/8d043cc0c34ef67fbbd5d9d9bc7242e1f44a85a0/api/geth_backend.go#L274-L279

### 2. First store received data, then call `receiveCallback`:

https://github.com/status-im/status-go/blob/7e6da26ed3f0611b6dec90cd5a8d8028d332e5a4/server/pairing/payload_receiver.go#L66-L75

This is required as well.
We were sending `received-account` event with `kdfIterations=0`, although we previously set it to `256000`.
This was happening, because the structure is updated during store procedure, which was happening after sending the event.

https://github.com/status-im/status-go/blob/7e6da26ed3f0611b6dec90cd5a8d8028d332e5a4/server/pairing/payload_receiver.go#L209-L212

### 3. Minor things

- Added `GethBackend.localPairing` flag, It  disables signalling `node.login` and `node.stopped` signals
- Added `Password` to `LocalPairingEvent`

I also rebased on top of f3c826275e1a634f4075156a054f9b29c7854df1, because latest status-desktop couldn't run without it. Probably makes sense to rebase the target branch as well before merging this.

### CODE REVIEW

**Comment by igor-sirotin** [comment]
@Samyoul @qfrank , I pushed some changes and updated PR description. Please, check it out.
With those updates manual sign in works fine in Desktop 👌

**Comment by qfrank** [comment]
let me test mobile based on this PR

**Comment by qfrank** [comment]
it broke mobile, we need align this on mobile before this PR merged cc @smohamedjavid @cammellos

File: server/pairing/payload_receiver.go [hunk 1]
```diff
@@ -94,7 +99,8 @@ func NewAccountPayloadReceiver(e *PayloadEncryptor, p *AccountPayload, config *R
 
 	return NewBaseBasePayloadReceiver(e, NewPairingPayloadMarshaller(p, l), aps,
```
Comment (qfrank) [inline server/pairing/payload_receiver.go]: would you mind also correct this name to `NewBasePayloadReceiver` ? thank you @igor-sirotin

**Review by qfrank — APPROVED** [review]
LGTM

File: server/pairing/payload_receiver.go (hunk 1 above)
Comment (igor-sirotin) [reply server/pairing/payload_receiver.go]: Fixed 👌

**Comment by smohamedjavid** [comment]
> it broke mobile, we need align this on mobile before this PR merged cc @smohamedjavid @cammellos

I'm working on changes for the mobile side. I need to test it out. I will update you, guys. 

Mobile PR [WIP]: https://github.com/status-im/status-mobile/pull/16065

**Comment by smohamedjavid** [comment]
@igor-sirotin and @qfrank - The mobile app is aligned with this PR. The pairing works as expected. No issues so far. 

I will open the PR (https://github.com/status-im/status-mobile/pull/16065) for reviews and QA testing. I will update once it's ready to merge.

**Comment by igor-sirotin** [comment]
@Samyoul can you review this one please?
It seems that it's ready on both mobile and desktop now.

I also changed the target branch directly to `master`, if that's fine.

File: server/pairing/payload_receiver.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ type BasePayloadReceiver struct {
 	receiveCallback func()
 }
 
-func NewBaseBasePayloadReceiver(e *PayloadEncryptor, um ProtobufUnmarshaller, s PayloadStorer, callback func()) *BasePayloadReceiver {
+func NewBasePayloadReceiver(e *PayloadEncryptor, um ProtobufUnmarshaller, s PayloadStorer, callback func()) *BasePayloadReceiver {
```
Comment (Samyoul) [inline server/pairing/payload_receiver.go:43]: 🤣

[Buttery Biscuit Base](https://www.youtube.com/watch?v=IfeyUGZt8nk)

image

**Review by Samyoul — APPROVED** [review]
Thank you for this work. Looks good.

**Pull request merged**